libraries/Microsoft.Bot.Builder.AI.TriggerTrees/Clause.cs

```diff
@@ -0,0 +1,305 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.AI.TriggerTrees
+{
+    public class Clause
+    {
+        public List<Expression> Predicates = new List<Expression>();
```

Comment:
As a convention, for public members I prefer exposing as properties. Also if it is readonly to external members so we get some more const-correctness i.e. change to 

public List<Expression> Predicates {get; <private> set} = new List<Expression>()